hadoop-ozone/common/src/main/java/org/apache/hadoop/ozone/protocolPB/OMPBHelper.java

```diff
@@ -84,15 +84,14 @@ public static OzoneAclInfo convertOzoneAcl(OzoneAcl acl) {
     default:
       throw new IllegalArgumentException("ACL type is not recognized");
     }
-    List<OzoneAclRights> aclRights = new ArrayList<>();
-
-    for (ACLType right : acl.getRights()) {
-      aclRights.add(OzoneAclRights.valueOf(right.name()));
-    }
+    List<OzoneAclRights> ozAclRights =
+        new ArrayList<>(acl.getAclBitSet().cardinality());
+    acl.getAclBitSet().stream().forEach(a -> ozAclRights.add(
+        OzoneAclRights.valueOf(ACLType.values()[a].name())));
 
```

Comment:
Many times, using normal for loops are much more efficient that lambda in Java. There is a huge perf cost. There was JIRA where Todd Lippcon replaced many lambdas in HDFS. so you might want to add a simple timer (start and stop) to make sure that these are equal in performance.

Reply:
@anuengineer  thanks for review. I remember that discussion, i think reason was too many type castings involved in in streams. I dont see any consistent approach to tackle it. (We are using streams all over the place) Updated other stream usage to foreach loop. In this particular case  foreach doesn't support Bitset so left it as it is. Hopefully performance of streams and parallel streams will improve in java.